server/app/services/PaginationSpec.java

```diff
@@ -1,23 +1,30 @@
 package services;
 
+import java.util.Optional;
+
 /** Specifies pagination behavior for a paginated query. */
 public class PaginationSpec {
 
   public static PaginationSpec MAX_PAGE_SIZE_SPEC = new PaginationSpec(Integer.MAX_VALUE, 1);
 
-  private final int pageSize;
-  private final int currentPage;
+  protected final int pageSize;
+  protected final Optional<Integer> currentPage;
```

Comment:
This is never used by the child so it doesn't need to be protected, and given it's never set by the child either this is likely another inheritance smell.

Reply:
n/a